tests/unit/cocoon/test_cocoon_printer.py

```diff
@@ -363,6 +383,33 @@ def test_format_holdings_response_success(
             for index, row in err.iterrows()
         ]
 
+    @parameterized.expand(
+        [
+            (
+                "standard_response",
+                responses,
+                2,
+                {"succ": ["ID00002", "ID00002"], "err": ["not found", "not found"]},
+            ),
+            ("empty_response", empty_response_with_full_shape, 0, {}),
+        ]
+    )
+    def test_format_reference_portfolios_response_success(
+        self, _, response, num_items, expected_value
+    ):
+        succ, err = format_reference_portfolios_response(response)
+        self.assertEqual(num_items, len(succ))
+        self.assertEqual(num_items, len(err))
+
+        [
```

Comment:
I know this was following the pattern before, but I think this ~can~ should be rewritten as:

```
for index, row in err.iterrows():
    self.assertEqual(expected_value["succ"][index], row[succ.columns[0]])
    self.assertEqual(expected_value["err"][index], row[err.columns[0]])
```